akka-distributed-data/src/main/scala/akka/cluster/ddata/ORMultiMap.scala

```diff
@@ -49,47 +46,50 @@ object ORMultiMap {
  */
 @SerialVersionUID(1L)
 final class ORMultiMap[A, B] private[akka] (
-  private[akka] val underlying: ORMap[A, ORSet[B]],
-  private[akka] val mapType:    Int)
+  private[akka] val underlying:      ORMap[A, ORSet[B]],
+  private[akka] val withValueDeltas: Boolean)
   extends DeltaReplicatedData with ReplicatedDataSerialization with RemovedNodePruning {
 
   override type T = ORMultiMap[A, B]
   override type D = ORMap.DeltaOp
 
   override def merge(that: T): T =
-    if (that.mapType == mapType) {
-      if (mapType == ORMultiMap.DeltaDeltaORMultiMapTag)
-        new ORMultiMap(underlying.mergeRetainingDeletedValues(that.underlying), mapType)
+    if (withValueDeltas == that.withValueDeltas) {
+      if (withValueDeltas)
+        new ORMultiMap(underlying.mergeRetainingDeletedValues(that.underlying), withValueDeltas)
       else
-        new ORMultiMap(underlying.merge(that.underlying), mapType)
-    } else throw new IllegalArgumentException("Trying to merge two ORMultiMaps of different mapType")
+        new ORMultiMap(underlying.merge(that.underlying), withValueDeltas)
+    } else throw new IllegalArgumentException("Trying to merge two ORMultiMaps of different map sub-type")
 
   /**
    * Scala API: All entries of a multimap where keys are strings and values are sets.
    */
-  def entries: Map[A, Set[B]] =
-    underlying.entries.filter { case (k, _) ⇒ underlying.keys.elements.contains(k) }.map { case (k, v) ⇒ k → v.elements }
+  def entries: Map[A, Set[B]] = if (withValueDeltas)
+    underlying.entries.collect { case (k, v) if underlying.keys.elements.contains(k) ⇒ k → v.elements }
+  else
+    underlying.entries.map { case (k, v) ⇒ k → v.elements }
 
   /**
    * Java API: All entries of a multimap where keys are strings and values are sets.
    */
   def getEntries(): java.util.Map[A, java.util.Set[B]] = {
     import scala.collection.JavaConverters._
     val result = new java.util.HashMap[A, java.util.Set[B]]
-    underlying.entries.filter { case (k, _) ⇒ underlying.keys.elements.contains(k) }.foreach {
-      case (k, v) ⇒ result.put(k, v.elements.asJava)
-    }
+    if (withValueDeltas)
+      underlying.entries.collect { case (k, v) if underlying.keys.elements.contains(k) ⇒ result.put(k, v.elements.asJava) }
```

Comment:
ah, this is actually not a `collect` since it's not returning anything

    underlying.entries.foreach { case (k, v) => if (underlying.keys.elements.contains(k)) result.put(k, v.elements.asJava) }